SwitchSpeak-iOS/SwitchSpeak-iOS/TouchGrid/Tree/Node.swift

```diff
@@ -47,7 +47,33 @@ class Node {
 		for childNode in self.childNodes {
 			childNode.unHighlightSubTree()
 		}
-	}	
+	}
+    
+    // it will retrun a node. if it has only undummy child node, it will return that child node, else it will return itself
+    func mergeNode() -> Node {
+        var unDummyNum = 0
+        var unDummyNode = Node()
+        for childNode in self.childNodes {
+            if childNode.dummy == false {
+                unDummyNum = unDummyNum + 1
+                unDummyNode = childNode
+            }
+        }
+        if unDummyNum == 1 {
+            return unDummyNode
+        }
+        else{
+            return self 
+        }
+    }
+    // it will make every child node shrink if its has only single undummy child node
+    func shrink() {
```

Comment:
Maybe change this to `shrinkChildren`? This function is modifying the children, not the node itself, so it could be clearer to stick 'children' in the title.